pkg/resources/scaleway/resources.go

```diff
@@ -0,0 +1,267 @@
+	c := cloud.(scaleway.ScwCloud)
+	servers, err := c.InstanceService().ListServers(&instance.ListServersRequest{
+		Zone: scw.Zone(c.Zone()),
+		Tags: []string{scaleway.TagClusterName + "=" + clusterName},
+	}, scw.WithAllPages())
+	if err != nil {
+		return nil, fmt.Errorf("failed to list servers: %s", err)
+	}
+
+	resourceTrackers := []*resources.Resource(nil)
+	for _, server := range servers.Servers {
+		resourceTracker := &resources.Resource{
+			Name: server.Name,
+			Type: resourceTypeServer,
+			ID:   server.ID,
+			Deleter: func(cloud fi.Cloud, tracker *resources.Resource) error {
+				return deleteServer(cloud, tracker)
+			},
+			Obj: server,
+		}
+		resourceTrackers = append(resourceTrackers, resourceTracker)
+	}
+
+	return resourceTrackers, nil
+}
+
+func listVolumes(cloud fi.Cloud, clusterName string) ([]*resources.Resource, error) {
+	c := cloud.(scaleway.ScwCloud)
+	volumesAll, err := c.InstanceService().ListVolumes(&instance.ListVolumesRequest{
+		Zone: scw.Zone(c.Zone()),
+	}, scw.WithAllPages())
+	if len(volumesAll.Volumes) < 1 {
+		return nil, nil
+	}
+	volumes, err := c.InstanceService().ListVolumes(&instance.ListVolumesRequest{
+		Zone: scw.Zone(c.Zone()),
+		Tags: []string{scaleway.TagClusterName + "=" + clusterName},
+	}, scw.WithAllPages())
+	if err != nil {
+		return nil, fmt.Errorf("failed to list volumes: %s", err)
+	}
+
+	resourceTrackers := []*resources.Resource(nil)
+	for _, volume := range volumes.Volumes {
+		resourceTracker := &resources.Resource{
+			Name: volume.Name,
+			Type: string(volume.VolumeType),
+			ID:   volume.ID,
+			Deleter: func(cloud fi.Cloud, tracker *resources.Resource) error {
+				return deleteVolume(cloud, tracker)
+			},
+			Obj: volume,
+		}
+		resourceTrackers = append(resourceTrackers, resourceTracker)
+	}
+
+	return resourceTrackers, nil
+}
+
+func deleteRecord(cloud fi.Cloud, tracker *resources.Resource) error {
+	c := cloud.(scaleway.ScwCloud)
+	recordDeleteRequest := &domain.UpdateDNSZoneRecordsRequest{
+		DNSZone: KopsDomainName,
+		Changes: []*domain.RecordChange{
+			{
+				Delete: &domain.RecordChangeDelete{
+					ID: &tracker.ID,
+				},
+			},
+		},
+	}
+	_, err := c.DomainService().UpdateDNSZoneRecords(recordDeleteRequest)
+	if err != nil {
+		return fmt.Errorf("failed to delete record %s: %s", tracker.Name, err)
+	}
+	return nil
+}
+
+func deleteLoadBalancer(cloud fi.Cloud, tracker *resources.Resource) error {
+	c := cloud.(scaleway.ScwCloud)
+	lbService := c.LBService()
+
+	err := lbService.DeleteLB(&lb.DeleteLBRequest{
+		Region: scw.Region(c.Region()),
+		LBID:   tracker.ID,
+	})
+	if err != nil {
+		return fmt.Errorf("failed to delete load-balancer %s: %s", tracker.ID, err)
+	}
+
+	return nil
+}
+
+func deleteServer(cloud fi.Cloud, tracker *resources.Resource) error {
+	c := cloud.(scaleway.ScwCloud)
+	zone := scw.Zone(c.Zone())
+	instanceService := c.InstanceService()
+
+	//_, err := instanceService.ServerAction(&instance.ServerActionRequest{
```

Comment:
Why is this code commented? Ensure that the server is in a state able to be deleted seems to be mandatory.